src/portable/portable.c

```diff
@@ -502,9 +502,9 @@ static int portable_extract_by_path(
         if (extract_os_release && !os_release)
                 return sd_bus_error_setf(error, SD_BUS_ERROR_INVALID_ARGS, "Image '%s' lacks os-release data, refusing.", path);
 
-        if (!extract_os_release && hashmap_isempty(unit_files))
-                return sd_bus_error_setf(error, SD_BUS_ERROR_INVALID_ARGS, "Couldn't find any matching unit files in image '%s', refusing.", path);
-
+        /* Note that we don't check here if we found any units, as we might be looking at
+         * a base OS layer, before opening the extension images. Let the caller do that
+         * after parsing all of them. */
```

Comment:
i'd drop the comment too, it's hard to grok why the comment is there unless one knows the history here. I'd drop the two lines without replacement